src/Dhall/Core.hs

```diff
@@ -1129,38 +1131,64 @@ subst _ _ (Embed p) = Embed p
     However, `normalize` will not fail if the expression is ill-typed and will
     leave ill-typed sub-expressions unevaluated.
 -}
-normalize :: Expr s a -> Expr t a
-normalize e = case e of
+normalize ::  Expr s a -> Expr t a
+normalize = normalizeWith (id)
```

Comment:
Fixed. I also applied this to other places where search/replace has ended up with extra parentheses.